src/auro-select.js

```diff
@@ -216,27 +216,41 @@ export class AuroSelect extends LitElement {
   }
 
   /**
-   * Adds the value string to the trigger.
-   * @param {string} str - The string to display in the trigger.
+   * Adds the value to the trigger.
+   * @param {object} option - The selected option to display it's value.
    * @private
    * @returns {void}
    */
-  updateDisplayedValue(str) {
+  updateDisplayedValue(option) {
     const dropdown = this.shadowRoot.querySelector('auro-dropdown');
     const triggerContentEl = dropdown.querySelector('#triggerFocus');
 
     // remove all existing rendered value(s)
+    triggerContentEl.querySelectorAll('auro-menuoption').forEach((elm) => {
+      elm.remove();
+    });
```

Comment:
```suggestion
    triggerContentEl.innerHTML = '';
```

We could use `innerHTML` to collect all the nodes within `triggerContentEl` and set to 'empty string'. And then we could delete 233-235 too.

Reply:
It appears that this has a negative side effect on the programmatic features.

Reply:
What do you mean this has a negative side effect?

Reply:
There is always a risk with using `innerHTML` and setting everything to an empty string is a little risky, that's all. 

You could also try to update this:

```js
    // remove all existing rendered value(s)
    triggerContentEl.querySelectorAll('auro-menuoption').forEach((elm) => {
      elm.remove();
    });

    triggerContentEl.querySelectorAll('[valuestr]').forEach((elm) => {
      elm.remove();
    });
``` 

to 

```js
    triggerContentEl.querySelectorAll('auro-menuoption, [valuestr]').forEach((elm) => {
      elm.remove();
    });
```